longevity_test.py

```diff
@@ -41,21 +42,37 @@ def test_custom_time(self):
                                     test_type=self.test_type,
                                     test_id=self.test_id)
         stress_queue = list()
+        write_queue = list()
 
         # prepare write workload
         prepare_write_cmd = self.params.get('prepare_write_cmd')
+        keyspace_num = self.params.get('keyspace_num', default=1)
+        pre_create_schema = self.params.get('pre_create_schema', default=False)
+
         if prepare_write_cmd:
-            write_queue = self.run_stress_thread(stress_cmd=prepare_write_cmd)
+            # If the test load is too heavy for one lader (e.g. many keyspaces), the load should be splitted evenly
+            # across the loaders (round_robin).
+            if pre_create_schema:
+                self._pre_create_schema()
+            if keyspace_num > 1 and self.params.get('round_robin', default='').lower() == 'true':
+                self.log.debug("Using round_robin for multiple Keyspaces...")
+                for i in xrange(1, keyspace_num):
+                    keyspace_name = 'keyspace{}'.format(i)
+                    write_queue.append(self.run_stress_thread(stress_cmd=prepare_write_cmd, keyspace_name=keyspace_name))
+                    time.sleep(5)
+            else:
+                write_queue = self.run_stress_thread(stress_cmd=prepare_write_cmd, keyspace_num=keyspace_num)
```

Comment:
`write_queue` should always be a list, you read it by for loop in line 67.

Reply:
good catch